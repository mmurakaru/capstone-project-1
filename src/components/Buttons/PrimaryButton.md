Regular Button with text

```js
<PrimaryButton disabled={false} handleClick={() => {}}>
  Save this song
</PrimaryButton>
```

```js
<PrimaryButton disabled={true} handleClick={() => {}}>
  Save this song
</PrimaryButton>
```
