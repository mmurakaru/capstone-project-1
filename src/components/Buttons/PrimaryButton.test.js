import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import PrimaryButton from './PrimaryButton'

describe('PrimaryButton', () => {
  it('renders correctly', () => {
    const { container, rerender } = render(
      <PrimaryButton disabled={false} handleClick={() => {}}>
        Save this song
      </PrimaryButton>
    )
    expect(container.firstChild).toMatchSnapshot()

    rerender(
      <PrimaryButton disabled={true} handleClick={() => {}}>
        Save this song
      </PrimaryButton>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
  it('calls handleClick with the correct id', () => {
    const handleClickMock = jest.fn()
    const { getByRole } = render(
      <PrimaryButton disabled={false} handleClick={() => handleClickMock('a')}>
        Save this song
      </PrimaryButton>
    )
    const button = getByRole('button')
    user.click(button)

    expect(handleClickMock).toHaveBeenCalledWith('a')
  })
})
