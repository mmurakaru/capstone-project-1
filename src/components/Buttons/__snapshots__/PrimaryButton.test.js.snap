// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton renders correctly 1`] = `
.c0 {
  background-color: transparent;
  border: var(--primary-dark) 2px solid;
  color: var(--primary-dark);
  font-size: 0.8em;
  padding: 6px;
  width: 100%;
}

.c0:disabled {
  opacity: 50%;
}

<button
  class="c0"
  data-testid="button"
>
  Save this song
</button>
`;

exports[`PrimaryButton renders correctly 2`] = `
<button
  class="PrimaryButton__ButtonStyled-sc-1jodd37-0 lQVev"
  data-testid="button"
  disabled=""
>
  Save this song
</button>
`;
